features/settingsFeature/src/main/java/com/panhuk/settingsfeature/SettingsFragment.kt

```diff
@@ -42,37 +40,35 @@ class SettingsFragment : Fragment() {
     container: ViewGroup?,
     savedInstanceState: Bundle?
   ): View {
-    _binding = FragmentSettingsBinding.inflate(inflater, container, false)
-    return binding.root
-  }
-
-  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-    super.onViewCreated(view, savedInstanceState)
-    binding.settingsFragment.setContent { CreateSettingsFragment() }
+    return ComposeView(requireContext()).apply {
+      setViewCompositionStrategy(DisposeOnViewTreeLifecycleDestroyed)
+      setContent {
+        Settings()
+      }
+    }
   }
 
-  override fun onDestroyView() {
+  override fun onDestroy() {
     viewModel.saveUsername()
```

Comment:
such functionality should be called from onPause() then it will be guaranteed saved in almost any case